app/controllers/circle/projects_controller.rb

```diff
@@ -61,6 +61,7 @@ def update
 
   def complete
     authorize! :complete, current_project
+    close_all_tasks_supplies
```

Comment:
Hi @NolanChan it would be better to move this method into the mutation Project::Complete. The mutation objects are intended for the business logic of lale (have a look at the gem description for details: https://github.com/cypriss/mutations). The controller should only mediate between request/model/mutation/view, but not contain business logic itself.